File: Selenium/Activities/Activity13.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://training-support.net/selenium/tables")
    print("Title is: ", driver.title)
    trEle = driver.find_elements(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr")
    print("No. of rows: ", len(trEle))
    tdEle = driver.find_elements(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr[1]/td")
    print("No. of columns: ", len(tdEle), "\n")
    secondRowEle = driver.find_elements(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr[3]")
    for item in secondRowEle:
        print(item.text, " ")
    print("\n", "2nd cell in 2nd row: ", driver.find_element(By.XPATH, "//table[contains(@class, 'striped')]/tbody/tr[2]/td[2]").text)
    driver.quit()

File: Selenium/Activities/Activity13.java
package Activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.WebDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;


public class Activity13 {
    public static void main(String[] args) throws InterruptedException
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver newDriver = new FirefoxDriver();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        newDriver.get("https://training-support.net/selenium/tables");
        System.out.println("Title is: "+newDriver.getTitle());
        List<WebElement> trEle = newDriver.findElements(By.xpath("//table[contains(@class,'striped')]/tbody/tr"));
        System.out.println("No. of rows is: " + trEle.size());
        List<WebElement> thirdEle = newDriver.findElements(By.xpath("//table[contains(@class,'striped')]/tbody/tr[3]/td"));
        System.out.println("Third row:");
        for(WebElement item : thirdEle)
            System.out.print(item.getText() + " ");
        System.out.println("\nContent in second row, second cell:"+newDriver.findElement(By.xpath("//table[contains(@class,'striped')]/tbody/tr[2]/td[2]")).getText());

        Thread.sleep(300);
        newDriver.quit();

    }
}
